src/pages/AppMarketplace/NFTunes/index.tsx

```diff
@@ -0,0 +1,253 @@
+
+        const arg = {
+          mvxNativeAuthOrigins: [window.location.origin],
+          mvxNativeAuthMaxExpirySeconds: 86400,
+          fwdHeaderMapLookup: {
+            "authorization": `Bearer ${tokenLogin.nativeAuthToken}`,
+          },
+          stream: true,
+        };
+        setCurrentIndex(index);
+
+        res = await dataNft.viewDataViaMVXNativeAuth(arg);
+        let blobDataType = BlobDataType.TEXT;
+
+        if (!res.error) {
+          if (res.contentType.search("application/json") >= 0) {
+            res.data = await (res.data as Blob).text();
+            res.data = JSON.stringify(JSON.parse(res.data), null, 4);
+          }
+        } else {
+          console.error(res.error);
+          toastError(res.error);
+        }
+        const viewDataPayload: ExtendedViewDataReturnType = {
+          ...res,
+          blobDataType,
+        };
+        setDataMarshalResponse(JSON.parse(res.data));
+
+        setViewDataRes(viewDataPayload);
+        setIsFetchingDataMarshal(false);
+      } else {
+        openModal();
+      }
+    } catch (err) {
+      console.error(err);
+      toastError((err as Error).message);
+      closeModal();
+      setIsFetchingDataMarshal(false);
+    }
+  }
+
+  return (
+    <HeaderComponent
+      pageTitle={"NF-Tunes"}
+      hasImage={true}
+      imgSrc={nfTunesBanner}
+      altImageAttribute={"NF-Tunes application"}
+      pageSubtitle={"Data NFTs that Unlock this App"}
+      dataNftCount={dataNfts.length}>
+      {dataNfts.length > 0 ? (
+        dataNfts.map((dataNft, index) => (
+          <DataNftCard
+            key={index}
+            index={index}
+            dataNft={dataNft}
+            isLoading={isLoading}
+            owned={flags[index]}
+            viewData={viewData}
+            modalContent={
+              isFetchingDataMarshal ? (
+                <div
+                  className="d-flex flex-column align-items-center justify-content-center"
+                  style={{
+                    minHeight: "40rem",
+                  }}>
+                  <div>
+                    <Loader noText />
+                    <p className="text-center text-foreground">
+                      {["ledger", "walletconnectv2", "extra"].includes(loginMethod) ? "Please sign the message using xPortal or Ledger" : "Loading..."}
+                    </p>
+                  </div>
+                </div>
+              ) : (
+                <>
+                  {viewDataRes && !viewDataRes.error && tokenLogin && currentIndex > -1 && (
+                    <AudioPlayer dataNftToOpen={dataNfts[currentIndex]} songs={dataMarshalResponse ? dataMarshalResponse.data : []} tokenLogin={tokenLogin} />
+                  )}
+                </>
+              )
+            }
+            modalTitle={"NF-Tunes"}
+            modalTitleStyle="p-4"
+          />
+        ))
+      ) : (
+        <h3 className="text-center text-white">No DataNFT</h3>
+      )}
+
+      {/* <Modal
```

Comment:
What's up with this huge comment? We shouldn't have these pushed further.

If we might need further code we can go back to git to search for it.